tests/controllers/storage.test.ts

```diff
@@ -602,6 +584,954 @@ describe('Storage controller', () => {
+          id: ''
+        },
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.deleteFolder(request, response);
+      } catch ({ message }) {
+        // Assert
+        expect(message).to.equal('Folder ID param is not valid');
+      }
+    });
+
+    it('should return error when execution fails', async () => {
+      // Arrange
+      const services = {
+        Folder: {
+          Delete: stubOf('Delete')
+            .rejects({
+              message: 'my-error'
+            }),
+        }
+      };
+      const controller = getController(services);
+      const request = getRequest({
+        behalfUser: {},
+        params: {
+          id: '1'
+        },
+        headers: {}
+      });
+      const sendSpy = sinon.spy();
+      const response = getResponse({
+        status: () => {
+          return {
+            send: sendSpy
+          };
+        }
+      });
+
+      // Act
+      await controller.deleteFolder(request, response);
+
+      // Assert
+      expect(services.Folder.Delete.calledOnce).to.be.true;
+      expect(sendSpy.calledOnce).to.be.true;
+      expect(sendSpy.args[0]).to.deep.equal([{
+        error: 'my-error'
+      }]);
+    });
+
+    it('should execute fine when no error', async () => {
+      // Arrange
+      const services = {
+        Folder: {
+          Delete: stubOf('Delete')
+            .resolves({
+              some: 'data'
+            }),
+        },
+        User: {
+          findWorkspaceMembers: stubOf('findWorkspaceMembers')
+            .resolves([
+              {}, {}
+            ]),
+        },
+        Notifications: {
+          folderDeleted: sinon.spy()
+        }
+      };
+      const controller = getController(services);
+      const request = getRequest({
+        behalfUser: {},
+        params: {
+          id: '1'
+        },
+        headers: {}
+      });
+      const sendSpy = sinon.spy();
+      const response = getResponse({
+        status: () => {
+          return {
+            send: sendSpy
+          };
+        }
+      });
+
+      // Act
+      await controller.deleteFolder(request, response);
+
+      // Assert
+      expect(services.Folder.Delete.calledOnce).to.be.true;
+      expect(services.User.findWorkspaceMembers.calledOnce).to.be.true;
+      expect(services.Notifications.folderDeleted.calledTwice).to.be.true;
+      expect(sendSpy.calledOnce).to.be.true;
+      expect(sendSpy.args[0]).to.deep.equal([{
+        some: 'data'
+      }]);
+    });
+
+  });
+
+  describe('Move folder', () => {
+
+    it('should fail if missing param `folderId`', async () => {
+      // Arrange
+      const controller = getController({});
+      const request = getRequest({
+        behalfUser: {},
+        body: {
+          folderId: '',
+          destination: '2'
+        },
+      });
+      const response = getResponse();
+
+      try {
+        // Act
+        await controller.moveFolder(request, response);
```

Comment:
as `chai` is being used here, in order to avoid false positives when a try-catch should happen, is advisable to force it to explicitly fail if this behavior does not happen. An example could be: adding an assertion that always fails after the line expected to `throw`:

```
await controller.moveFolder(request, response);
expect(true).to.be.false;
```